src/tlv/TlvObject.ts

```diff
@@ -80,6 +81,15 @@ export class ObjectSchema<F extends TlvFields> extends TlvSchema<TypeFromFields<
             const { field, name } = fieldName;
             result[name] = field.schema.decodeTlvInternalValue(reader, elementTypeLength);
         }
+        // Check mandatory fields and, if missing, populate with fallback value if defined.
+        for (const name in this.fieldDefinitions) {
```

Comment:
I'm confused why this logic is under OptionalFieldType as opposed to FieldType, especially given the comment that it only applies to mandatory fields.

Reply:
Are you sure that not GitHub tricked you? ;-) It is in the "ObjectsSchema" which is generic

Reply:
oh right, my bad.  thanks!